### ISSUE TITLE
Crash when deployed on Linux (Ubuntu, AWS, Heroku)

### ISSUE THREAD

**Issue by miroslavkovac** [vapor/mongo-provider#6]
I am experiencing a strange crash with when working with Vapor+Mongo. I have an app deployed on Heroku with just one route that does
```
do {
    let objects = try MyEntity.all()
    return "done"
} catch (let error) {
    throw Abort.custom(status: .badRequest, message: error.localizedDescription)
}
```
Notice, that after fetching all entities, they are discarded.

`MyEntity` conforms to `Model` and only thing that it extracts is the id.

This works fine when running from the Xcode (I have hardcoded a connection URL to mLab DB in main.swift, so that I have least moving parts). When I deploy it to Heroku, the first request succeeds and after that the server crashes (without any logs or stack trace) and I have to restart it.

The mongo collection that I am fetching contains only a single object that has only the _id and a date:
```
{
    "_id" : ObjectId("58223184aa892cc027c91456"),
    "date" : ISODate("2016-10-05T16:54:36.705Z")
}
```
Now, the interesting part is that if I remove the `date` then all requests succeed on both Xcode and Heroku, but if I put the `date` back, it still works fine from Xcode, but starting from second/third request onwards, it starts crashing on Heroku.

**Comment by tanner0101** [comment]
Set Vapor to `debug` environment and check the logs. It's likely that extracting the date is failing. Models that fail to initialize are ignored.

**Comment by miroslavkovac** [comment]
Yes, I am aware that the date is still unsupported. I can confirm that when running from Xcode in `debug` environment:

```
[FluentMongo] Could not convert BSON to Node: dateTime(2016-10-05 16:14:03 +0000)
```

But I am wondering why is it crashing whole server when deployed on Heroku? Even if I use `--env=production` in Xcode, it works fine - without any crash or fatal error, it just skips that date.

I think that problem is somewhere else, because everything is surrounded with `do-catch`, so at least it should return `.badRequest` and not crash.

Is there any way of getting a crash log or stack trace when already deployed an run on Heroku? The best would be to attach a debugger.. :D

**Comment by tanner0101** [comment]
Didn't realize the whole process was crashing. @LoganWright might know how to get stack traces from Heroku, I'm not too familiar myself.

If we got the stack trace that would help out a ton. Are you using anything from `Foundation`? That has been known to cause crashes.

**Comment by miroslavkovac** [comment]
Well, whole Vapor and MongoDriver are using Foundation as from what I can tell :)

It is not limited to Heroku (I will update the issue title). It also happens when deployed on AWS and locally on Linux.

**Comment by miroslavkovac** [comment]
Just to give an update and close the issue.

The crash was caused by a segmentation fault that happens when calling a `description` function on `Foundation.Date` object for the second time (https://bugs.swift.org/browse/SR-2485).

Since `BSON.Value..dateTime` is unsupported in MongoDriver, we will try to print error message about unsupported enum case, causing the print of associated `date`.

I have opened a PR to remove the printing of unsupported BSON Value (https://github.com/vapor/mongo-driver/pull/23) and lets hope Apple fixes SR-2485 soon.
